Content.Server/Station/Systems/StationJobsSystem.cs

```diff
@@ -0,0 +1,424 @@
+    /// <param name="stationJobs">Resolve pattern, station jobs component of the station.</param>
+    /// <returns>Whether or not assignment was a success.</returns>
+    /// <exception cref="ArgumentException">Thrown when the given station is not a station.</exception>
+    public bool TryAssignJob(EntityUid station, string jobPrototypeId, StationJobsComponent? stationJobs = null)
+    {
+        return AdjustJobSlots(station, jobPrototypeId, -1, false, stationJobs);
+    }
+
+    /// <inheritdoc cref="AdjustJobSlots(Robust.Shared.GameObjects.EntityUid,Content.Shared.Roles.JobPrototype,int,bool,Content.Server.Station.Components.StationJobsComponent?)"/>
+    /// <param name="station">Station to adjust the job slot on.</param>
+    /// <param name="job">Job to adjust.</param>
+    /// <param name="amount">Amount to adjust by.</param>
+    /// <param name="createSlot">Whether or not it should create the slot if it doesn't exist.</param>
+    /// <param name="stationJobs">Resolve pattern, station jobs component of the station.</param>
+    public bool AdjustJobSlots(EntityUid station, JobPrototype job, int amount, bool createSlot = false, StationJobsComponent? stationJobs = null)
+    {
+        return AdjustJobSlots(station, job.ID, amount, createSlot, stationJobs);
+    }
+
+    /// <summary>
+    /// Attempts to adjust the given job slot by the amount provided.
+    /// </summary>
+    /// <param name="station">Station to adjust the job slot on.</param>
+    /// <param name="jobPrototypeId">Job prototype ID to adjust.</param>
+    /// <param name="amount">Amount to adjust by.</param>
+    /// <param name="createSlot">Whether or not it should create the slot if it doesn't exist.</param>
+    /// <param name="stationJobs">Resolve pattern, station jobs component of the station.</param>
+    /// <returns>Whether or not slot adjustment was a success.</returns>
+    /// <exception cref="ArgumentException">Thrown when the given station is not a station.</exception>
+    public bool AdjustJobSlots(EntityUid station, string jobPrototypeId, int amount, bool createSlot = false,
+        StationJobsComponent? stationJobs = null)
+    {
+        if (!Resolve(station, ref stationJobs))
+            throw new ArgumentException("Tried to use a non-station entity as a station!", nameof(station));
+
+        var jobList = stationJobs.JobList;
+
+        // This should:
+        // - Return true when zero slots are added/removed.
+        // - Return true when you add.
+        // - Return true when you remove and do not exceed the number of slot available.
+        // - Return false when you remove from a job that doesn't exist.
+        // - Return false when you remove and exceed the number of slots available.
+        // And additionally, if adding would add a job not previously on the manifest when createSlot is false, return false and do nothing.
+        switch (jobList.ContainsKey(jobPrototypeId))
+        {
+            case false when amount < 0:
```

Comment:
this PR uses literally every piece of c# syntax. didnt even know you could do this on normal switch cases